pkg/cloud/azure/actuators/machine/actuator_test.go

```diff
@@ -521,6 +523,59 @@ func TestAvailabilityZones(t *testing.T) {
 	}
 }
 
+func TestGetZone(t *testing.T) {
+	testCases := []struct {
+		inputZone *string
+		expected  string
+	}{
+		{
+			inputZone: nil,
+			expected:  "",
+		},
+		{
+			inputZone: pointer.StringPtr("3"),
+			expected:  "3",
+		},
+		{
+			inputZone: pointer.StringPtr(""),
```

Comment:
nil? or ""?

Reply:
nil is the first case, this is intentionally "" to validate any available zone is given